release-notes/unreleased/2049.bug.rst

```diff
@@ -0,0 +1,2 @@
+Pin ``pydantic`` dependency version to ``<2.10`` to avoid a regression breaking
```

Comment:
No need to release to get out this hotfix. But I figure it's worth documenting because it may impact end users that they can't use Pydantic <2.10. 

It's somewhat of a gray area whether this is overkill and noisy; because Pydantic is an important dependency people usually directly depend on it, I lean towards documenting it.